src/Symfony/Component/HttpFoundation/Session/Storage/Handler/AbstractSessionHandler.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+abstract class AbstractSessionHandler implements \SessionHandlerInterface, \SessionUpdateTimestampHandlerInterface
```

Comment:
i think this class can be implemented as a proxyhandler as well similar to SessionHandlerWrapper that wraps another SessionHandlerInterface. just check that the wrapped handler  does not implement SessionUpdateTimestampHandlerInterface. or alternatively, in case it implements SessionUpdateTimestampHandlerInterface just proxy those method as well. that might even be better because this class `destroy` logic might be useful even for those handlers.

Reply:
this way, the other session handlers don't need to be changed and they don't need to know about `$this->sessionName`

Reply:
(re-posting answer on the correct comment)
The issue with going with only a proxy is that it will prevent skipping the lock in PdoSessionHandler, because we couldn't know if we're having the required logic or not.
Oh, and implementating updateTimestamp per-handler would have no contract to be based on (and adding one would be worse.)

Reply:
I don't see the problem with the PdoSessionHandler. Instead of AbstractSessionHandler::doRead and doWrite etc, just call `$this->wrapperSessionHandler->read()`

Reply:
And for updateTimestamp: 
```
if ($this->wrapperSessionHandler instanceof SessionUpdateTimestampHandlerInterface) {
   $this->wrapperSessionHandler ->updateTimestamp();
} else {
    $this-save();
}
```

Reply:
> And for updateTimestamp:

that implies implementing `SessionUpdateTimestampHandlerInterface` on existing handlers, so would require implementing also "validateId". That's what the abstract is for.

Reply:
validateId can be an empty method as it is not called. SessionUpdateTimestampHandlerInterface is implemented in exsting handlers anyway by extending the abstract. another advantage is that this proxy is usable with any existing session handlers even when they not come from symfony.

Reply:
I don't see any benefit to removing the abstract. Creating dummy functions is not better, what would this improve?
SessionHandlersWrapper is already usable with any existing handler, so this need is already filled.
What are you seeking for?

Reply:
So it only needs one wrapper. Having both an abstract and a wrapper for no reason just makes the api confusing. Also the naming of these classes should be improved. abstract and wrapper just does not say anything about what they achieve which should be clear from the class name.

Reply:
> going with only a proxy is that it will prevent skipping the lock in PdoSessionHandler

The issue here is that the handler (pdo in this case) will not be able to make the assumption that session ids are unique. Thus, we would not be able to skip the locking logic in PDO.

I really don't see the point of not having the abstract for the sake of it sorry The wrapper is needed to add the strict behavior to the native proxy (too bad it doesn't have it already), and is useful for decorating existing handlers of course. About the name, suggestions welcomed (naming is hard :) )

Reply:
> validateId can be an empty method as it is not called

as soon as `SessionUpdateTimestampHandlerInterface` is implemented, it *can* be called, unless we want to provide deficient implementation that must be wrapped. That's why the abstract is needed: to provide a working implement of it.

Reply:
This class needs phpdoc to explain what it achieves.